source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -377,9 +416,104 @@ TEST_F(nifgenDriverApiTest, ResetInterchangeCheck_ResetsSuccessfully)
+  request.set_trigger(fgen::Trigger::TRIGGER_NIFGEN_VAL_SCRIPT_TRIGGER);
+  request.set_trigger_id("ScriptTrigger0");
+  fgen::SendSoftwareEdgeTriggerResponse response;
+  ::grpc::Status status = GetStub()->SendSoftwareEdgeTrigger(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
 }
 
+TEST_F(NiFgenDriverApiTest, AllocateNamedWaveform_WriteNamedWaveformF64_WaveformWrittenSuccessfully)
+{
+  std::string channel_name = "0";
+  std::string waveform_name = "TestWaveform";
+  int waveform_size = 5;
+  set_int32_attribute("", fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_OUTPUT_MODE, fgen::OutputMode::OUTPUT_MODE_NIFGEN_VAL_OUTPUT_SCRIPT);
+  ::grpc::ClientContext alloc_wfm_context;
+  fgen::AllocateNamedWaveformRequest alloc_wfm_request;
+  alloc_wfm_request.mutable_vi()->set_id(GetSessionId());
+  alloc_wfm_request.set_channel_name(channel_name);
+  alloc_wfm_request.set_waveform_name(waveform_name);
+  alloc_wfm_request.set_waveform_size(waveform_size);
+  fgen::AllocateNamedWaveformResponse alloc_wfm_response;
+  ::grpc::Status alloc_wfm_status = GetStub()->AllocateNamedWaveform(&alloc_wfm_context, alloc_wfm_request, &alloc_wfm_response);
+  EXPECT_TRUE(alloc_wfm_status.ok());
+  expect_api_success(alloc_wfm_response.status());
+  
+  double waveform[] = {1.55, 40.4, 21.6, 0.7, 15.89};
+  ::grpc::ClientContext write_wfm_context;
+  fgen::WriteNamedWaveformF64Request write_wfm_request;
+  write_wfm_request.mutable_vi()->set_id(GetSessionId());
+  write_wfm_request.set_channel_name(channel_name);
+  write_wfm_request.set_waveform_name(waveform_name);
+  write_wfm_request.mutable_data()->CopyFrom(google::protobuf::RepeatedField<double>(waveform, waveform + waveform_size));
+  fgen::WriteNamedWaveformF64Response write_wfm_response;
+  ::grpc::Status write_wfm_status = GetStub()->WriteNamedWaveformF64(&write_wfm_context, write_wfm_request, &write_wfm_response);
+
+  EXPECT_TRUE(write_wfm_status.ok());
+  expect_api_success(write_wfm_response.status());
+}
+
+TEST_F(NiFgenDriverApiTest, ExportTriggerMode_ResetAndImportConfiguration_TriggerModeConfigurationIsImported)
+{
+  const char* channel_name = "0";
+  fgen::TriggerMode expected_value = fgen::TriggerMode::TRIGGER_MODE_NIFGEN_VAL_SINGLE;
+  configure_trigger_mode(channel_name, expected_value);
+  auto exported_configuration_response = export_attribute_configuration_buffer();
+
+  reset();
+  import_attribute_configuration_buffer(exported_configuration_response);
+
+  double set_value = get_int32_attribute("0", fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_TRIGGER_MODE);
+  EXPECT_EQ(expected_value, set_value);
+}
+
+TEST_F(NiFgenDriverApiTest, AllocateWaveform_WriteWaveformComplexF64_WaveformWrittenSuccessfully)
+{
+  std::string channel_name = "0";
+  int waveform_size = 1;
+   set_int32_attribute("", fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_OUTPUT_MODE, fgen::OutputMode::OUTPUT_MODE_NIFGEN_VAL_OUTPUT_SEQ);
+  ::grpc::ClientContext alloc_wfm_context;
```

Comment:
Move `AllocateWaveform`, `WriteWaveformComplexF64` and corresponding `request` setup to helper methods. (same as previous comment)